MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -5159,6 +5162,19 @@ public string GetHwTonemapFilter(EncodingOptions options, string hwTonemapSuffix
                 overlayFilters.Add("overlay_videotoolbox=eof_action=pass:repeatlast=0");
             }
 
+            var needFiltering = mainFilters.Any(f => !string.IsNullOrEmpty(f)) ||
+                                subFilters.Any(f => !string.IsNullOrEmpty(f)) ||
+                                overlayFilters.Any(f => !string.IsNullOrEmpty(f));
+
+            // This is a workaround for ffmpeg's hwupload implementation
+            // For VideoToolbox encoders, a hwupload without a valid filter actually consuming its frame
+            // will cause the encoder to produce incorrect frames.
+            if (needFiltering)
+            {
+                hwuploadFilters.AddRange(mainFilters);
+                mainFilters = hwuploadFilters;
```

Comment:
```suggestion
                mainFilters.Insert(0, "hwupload=derive_device=videotoolbox");
                mainFilters.Insert(0, "format=nv12|p010le|videotoolbox_vld");
```

If order doesn't matter just use `Add()`, and remove `hwuploadFilters`

Reply:
It matters. `hwupload*` should be the first.

Reply:
Yes, hwupload has to be first.

Reply:
Ok, my suggestion is to not create a list just to maybe use it- but to just insert the required filters at the start of the list when needed

Reply:
Done.